processor/src/lib.rs

```diff
@@ -122,36 +139,65 @@ pub fn execute_iter(program: &Program, inputs: &ProgramInputs) -> VmStateIterato
 // PROCESS
 // ================================================================================================
 
-pub struct Process {
+pub struct Process<ADV>
+where
+    ADV: AdviceProvider,
+{
     system: System,
     decoder: Decoder,
     stack: Stack,
     range: RangeChecker,
     chiplets: Chiplets,
-    advice: AdviceProvider,
+    advice: ADV,
 }
 
-impl Process {
+impl<ADV> Process<ADV>
+where
+    ADV: AdviceProvider,
+{
     // CONSTRUCTORS
     // --------------------------------------------------------------------------------------------
-    /// Creates a new process with the provided inputs.
-    pub fn new(kernel: &Kernel, inputs: ProgramInputs) -> Self {
-        Self::initialize(kernel, inputs, false)
+
+    /// Creates a new process with the provided stack inputs.
+    pub fn new<STK, V>(kernel: &Kernel, advice: ADV, stack_init: STK) -> Self
+    where
+        STK: IntoIterator<Item = V>,
+        V: Into<Felt>,
```

Comment:
I'm not sure we need to change the signature this much as we'll probably have a dedicated struct type for `stack_init`.

Reply:
Done! We now split it into several steps and can be merged incrementally to `next`